### ISSUE TITLE
Build fails on Mac because .cargo/config.toml has CPU architecture set to haswell

### ISSUE THREAD

**Issue by nastynaz** [serenity-rs/serenity#2872]
Problem: running the examples or otherwise trying to build this repo fails on my machine (Mac M1, Apple Silicon).

The error message is:
```
'haswell' is not a recognized processor for this target (ignoring processor)
```

The issue seems to be because of this line: https://github.com/serenity-rs/serenity/blob/4e5189c5afbd48005eb6b77d916182da103437d7/.cargo/config.toml#L2

Once I changed it to "target-cpu=native" it worked.

**Comment by GnomedDev** [comment]
We can remove this file entirely now, as we don't use simd-json anymore.

**Comment by GnomedDev** [comment]
This is fixed on the `next` branch, which will be released as serenity 0.13.
